lib/usb/usb_process.js

```diff
@@ -1,236 +1,271 @@
-      // Tell the daemon that we have finished writing to the control stream
-      daemon.write(protocol.controlClose(self.id));
-      // Call the callback
-      callback && callback();
-    }});
-
-    if (self.control.credit) {
-      self._drainBackPressure(self.control, protocol.controlWrite);
+  // Checks if this process is ready to be closed
+  // and closes it if it can
+  self.closeProcessIfReady = function() {
+    // If this process hasn't been closed before
+    // and all of the streams are closed
+    if (self.active && self.stdout.closed
+            && self.stderr.closed
+            && self.stdin.closed
+            && self.control.closed) {
+      // Close this process
+      self.close();
     }
   };
 
-  // Intercepting write to the stdin stream
-  this.stdin._write = function(chunk, enc, callback) {
-    // Add this chunk to the backpressure array
-    self.stdin.backPressure.push({buffer: chunk, callback: callback});
-    if (self.stdin.credit) {
-      self._drainBackPressure(self.stdin, protocol.stdinWrite);
-    }
-  };
+  // When any of the streams complete, check if the process is ready to close
+  self.stdin.once('finish', self.closeProcessIfReady);
+  self.stdout.once('end', self.closeProcessIfReady);
+  self.stderr.once('end', self.closeProcessIfReady);
 
-  this._drainBackPressure = function(stream, genFunc) {
-        // If we can write data to the remote pipe and we have data to write
-    while (stream.credit && stream.backPressure.length) {
-      // Get the first entry
-      var entry = stream.backPressure[0];
-      // Set the data to write to the daemon
-      var chunk = entry.buffer;
-
-      // If we are attempting to write more bytes than the socket can handle
-      if (chunk.length > stream.credit) {
-        // Slice the chunk that will be sent
-        chunk = chunk.slice(0, stream.credit);
-      }
-
-      // Cut that off the front of the remaining bytes in the entry
-      entry.buffer = entry.buffer.slice(chunk.length);
-
-      // Split the chunk into smaller packets
-      var indices = self._packetize(MAX_PACKET_SIZE, chunk);
-
-      // For each packet
-      indices.forEach(function(index) {
-        // Slice out the data packet
-        var buffer = chunk.slice(index.start, index.end);
-        // Send the header for our control write
-        daemon.write(genFunc(self.id, buffer.length));
-        // Send over the data to the control stream
-        daemon.write(buffer);
-      });
-
-      // If we sent all of the remaining bytes of this write
-      if (entry.buffer.length == 0) {
-        // Remove it from back pressure
-        stream.backPressure.shift();
-        // Call the callback if applicable
-        entry.callback && entry.callback();
-      }
+  // This function is primarily intended for compatibility with the ssh2 stream API
+  self.stdout.signal = function(signalToSend) {
+    switch(signalToSend) {
+      case 'KILL':
```

Comment:
I'd expect to see 'INT' here, as that's the signal you should be sending first. (I'm thinking of `tessel run` ctrl-c in particular)